Add script for batch updating study roles

Ruborcalor opened the pull request:
Real time role updates are taken care of in #13. This pull request adds:
- Pipfile
- Updates fetch_user_stats to no longer try and pull streak information from v1 because v1 doesn't properly handle streaks anymore
- Adds `batch_update_roles.py` which can be run to update all the roles of all the users in the discord server according to v2 data. It shouldn't be needed frequently because of realtime role updates, but if many roles were ever to get out of sync, this script could be used.

Zackhardtoname commented:
Thanks for working on this!
It looks like it's unfinished? In particular, `sqlalchemytest.py` has incomplete statements.

Zackhardtoname commented:
I incorporated the ideas into the recent role updates for #13 
I would love to merge this one in though for the other ideas like pipefile and tests. When you are done, could you kindly add a description of the changes? Thanks!

Ruborcalor commented:
Nice stuff with #13, definitely the better approach!

avramkjiprovski commented:
I believe that both of these issues are fixed now. I'm documenting this here for the sake of documenting events and because it seems to me that it was related to this issue thread.

Some members saying that their roles haven't updated as well counter bot taking too long to return `!me` output in command spam channels.

Ruborcalor commented:
> I incorporated the ideas into the recent role updates for #13
> I would love to merge this one in though for the other ideas like pipefile and tests. When you are done, could you kindly add a description of the changes? Thanks!

I updated the description of the changes:

Real time role updates are taken care of in #13. This pull request adds:
- Pipefile
- Updates fetch_user_stats to no longer try and pull streak information from v1 because v1 doesn't properly handle streaks anymore
- Adds `batch_update_roles.py` which can be run to update all the roles of all the users in the discord server according to v2 data. It shouldn't be needed frequently because of realtime updates role updates, but if many roles were ever to get out of sync, this script could be used.

Zackhardtoname commented:
> I believe that both of these issues are fixed now. I'm documenting this here for the sake of documenting events and because it seems to me that it was related to this issue thread.
> 
> Some members saying that their roles haven't updated as well counter bot taking too long to return `!me` output in command spam channels.

Yeah at first I was issuing API calls even if the role I calculated to add was already assigned before and hit rate-limiting from discord. be658d6 should have fixed that problem.
